src/sections/dataset/dataset-files/dataset-upload-files-button/DatasetUploadFilesButton.tsx

```diff
@@ -0,0 +1,27 @@
+import { Button } from '@iqss/dataverse-design-system'
+import { PlusLg } from 'react-bootstrap-icons'
+import { useSession } from '../../../session/SessionContext'
+import styles from './DatasetUploadFilesButton.module.scss'
+import { useTranslation } from 'react-i18next'
+
+export function DatasetUploadFilesButton() {
+  const { t } = useTranslation('dataset')
+  const { user } = useSession()
+  const userHasDatasetUpdatePermissions = true // TODO - Implement permissions
+  const datasetLockedFromEdits = false // TODO - Ask Guillermo if this a dataset property coming from the api
```

Comment:
Locks are exposed via API endpoint: https://guides.dataverse.org/en/latest/api/native-api.html?highlight=locks

But I'm not completely sure if we are talking about the same kind of locks. Can you send to me the JSF logic you are replicating here?

Let's take note of all these TODOes to create issues at the end of the sprint as last time.